faunadb/objects.py

```diff
@@ -16,44 +15,46 @@ class Ref(_Expr):
   Queries that require a Ref will not work if you just pass in a string.
   """
 
-  def __init__(self, *parts):
-    """
-    Create a Ref from a string, such as ``Ref("databases/prydain")``.
-    Can also call ``Ref("databases", "prydain")`` or ``Ref(Ref("databases"), "prydain")``.
-    """
-    parts_str = "/".join(str(part) for part in parts)
+  def __init__(self, id, cls=None, db=None):
+    if id is None:
+      raise ValueError("The Ref must have an id.")
+
+    value = {"id": id}
+
+    if cls != None:
+      value["class"] = cls
 
-    super(Ref, self).__init__(parts_str)
+    if db != None:
+      value["database"] = db
 
-  def to_class(self):
+    super(Ref, self).__init__(value)
+
+  def class_(self):
     """
     Gets the class part out of the Ref.
-    This is done by removing the id.
-    So ``Ref("a", "b/c").to_class()`` will be ``Ref("a/b")``.
     """
-    parts = self.value.split("/")
-    if len(parts) == 1:
-      return self
-    return Ref(*parts[:-1])
+    return self.value.get("class")
+
+  def database(self):
+    """
+    Gets the database part out of the Ref.
+    """
+    return self.value.get("database")
 
   def id(self):
     """
-    Removes the class part of the Ref, leaving only the id.
-    This is everything after the last ``/``.
+    Gets the id part out of the Ref.
     """
-    parts = self.value.split("/")
-    if len(parts) == 1:
-      raise ValueError("The Ref does not have an id.")
-    return parts[-1]
+    return self.value["id"]
 
   def to_fauna_json(self):
     return {"@ref": self.value}
 
   def __str__(self):
-    return self.value
+    return "Ref(%s)" % self.value
```

Comment:
This would be a weird representation. I'd prefer tot see all fields as something like: `Ref(id=1, class=foo, database=bar)`.